src/bindings/jna/libelektra4j/src/main/java/org/libelektra/plugin/PropertiesStorage.java

```diff
@@ -1,74 +0,0 @@
-package org.libelektra.plugin;
```

Comment:
Why is this plugin removed here? The PR is vey difficult to read due to reformatting of all files. Please describe some formatting for Java in doc/CODING.md (can be separate to Groovy, or also reformat scripts/jenkins/Jenkinsfile)

Reply:
I did not reformat anything. I deleted a lot of files which basically had no senseful meaning like the Echo.java or the PropertiesStorage.java because:

1. I want to use the native elektra plugin from is implemented by the new Plugin class and used by the ErrorCodeTest (I use the `error` plugin there)
2. This would be in a conflicting name with the current `Plugin` interface which more or less just gave 4 methods to implement and basically had no real connection to KDB at all
3. The "implemented Plugins" in Java did not interact with KDB at all except for reading something from the parent key. Isn't it the meaning of elektra to load all into a keyset and handle the operations there rather than doing it with the native java code that directly manipulates the files? The specification for example could never be used with a java plugin. 

Due to all of these reasons I decided to remove them because they might even mislead new developers in implementing plugins in java

Reply:
> I did not reformat anything.

Please look at the diff in GitHub. E.g. src/bindings/jna/libelektra4j/src/main/java/org/libelektra/Plugin.java had tabs before and in this PR it now uses spaces.

> The "implemented Plugins" in Java did not interact with KDB at all except for reading something from the parent key. Isn't it the meaning of elektra to load all into a keyset and handle the operations there rather than doing it with the native java code that directly manipulates the files? The specification for example could never be used with a java plugin.

These plugins are actual Elektra plugins which can be mounted (and used as storage plugin). Please read src/plugins/jni/README.md to understand the plugin interface.

I am very glad if you improve the plugin interface (even beyond the introduction of the new error concept) but:

- there needs to be documentation on how to use it
- you need to update existing documentation (this PR does not touch documentation at all)
- there need to be examples
- there need to be test cases

Reply:
> Please look at the diff in GitHub. E.g. src/bindings/jna/libelektra4j/src/main/java/org/libelektra/Plugin.java had tabs before and in this PR it now uses spaces

On my GitHub I do not see a difference between tabs and spaces, do you have a special plugin or setting which shows this?

Apart from that: It is the default setting in intellij is to use spaces and it sometimes even suggests to use them instead of tabs if it sees it is used in a file. It is in general much more common to use spaces in java. I can revert that back to tabs if you like but imho it does not make a lot of difference at all

> These plugins are actual Elektra plugins which can be mounted (and used as storage plugin). Please read src/plugins/jni/README.md to understand the plugin interface.

I readded them and renamed the plugin to nativeplugin which was used by myself to for the test case.

Reply:
You see tabs/spaces by trying to mark them.

It is commons sense that mixing tabs and spaces is the worst of all possibilities. So yes, please revert to tabs. We can convert everything to spaces in a later PR to have more Java-style. In any case, it is good if this is documented in doc/CODING.md. Even better would be to automatically reformat with a tool (and on the build server).

Reply:
> You see tabs/spaces by trying to mark them.

Honestly I have seen problem with line endings but never with tabs/spaces. Nontheless I think there is a good plugin in case you are using firefox or chrome for that https://github.com/glebm/render-whitespace-on-github. I used it now to quickly see tabs/spaces.

I fixed the tab/spaces stuff. Please review the rest of the PR :)